tools/slicec-cs/src/builders.rs

```diff
@@ -366,34 +363,26 @@ impl FunctionBuilder {
             );
         }
 
-        match context {
-            TypeContext::IncomingParam => {
-                self.add_parameter(
-                    "IceRpc.Features.IFeatureCollection",
-                    &escape_parameter_name(&parameters, "features"),
-                    None,
-                    Some("The dispatch features.".to_owned()),
-                );
-            }
-            TypeContext::OutgoingParam => {
-                self.add_parameter(
-                    "IceRpc.Features.IFeatureCollection?",
-                    &escape_parameter_name(&parameters, "features"),
-                    Some("null"),
-                    Some("The invocation features.".to_owned()),
-                );
-            }
-            _ => panic!("Unexpected context value"),
+        if is_dispatch {
+            self.add_parameter(
+                "IceRpc.Features.IFeatureCollection?",
+                &escape_parameter_name(&parameters, "features"),
+                Some("null"),
+                Some("The invocation features.".to_owned()),
+            );
+        } else {
+            self.add_parameter(
+                "IceRpc.Features.IFeatureCollection",
+                &escape_parameter_name(&parameters, "features"),
+                None,
+                Some("The dispatch features.".to_owned()),
+            );
```

Comment:
This looks backwards.

Reply:
IMO using a bool makes this ever error prone. What about using a two discriminant enum?

Reply:
Yes, I picked a better name for this bool. I was too mechanistic in my translation.